gramps/gui/clipboard.py

```diff
@@ -1627,75 +1573,13 @@ def edit_obj(self, objclass, handle):
-            if person:
-                try:
-                    EditPerson(self.dbstate,
-                               self.uistate, [], person)
-                except WindowActiveError:
-                    pass
-        elif objclass == 'Event':
-            event = self.dbstate.db.get_event_from_handle(handle)
-            if event:
-                try:
-                    EditEvent(self.dbstate,
-                              self.uistate, [], event)
-                except WindowActiveError:
-                    pass
-        elif objclass == 'Family':
-            ref = self.dbstate.db.get_family_from_handle(handle)
-            if ref:
-                try:
-                    EditFamily(self.dbstate,
-                               self.uistate, [], ref)
-                except WindowActiveError:
-                    pass
-        elif objclass == 'Source':
-            ref = self.dbstate.db.get_source_from_handle(handle)
-            if ref:
-                try:
-                    EditSource(self.dbstate,
-                               self.uistate, [], ref)
-                except WindowActiveError:
-                    pass
-        elif objclass == 'Place':
-            ref = self.dbstate.db.get_place_from_handle(handle)
-            if ref:
-                try:
-                    EditPlace(self.dbstate,
-                               self.uistate, [], ref)
-                except WindowActiveError:
-                    pass
-        elif objclass == 'Repository':
-            ref = self.dbstate.db.get_repository_from_handle(handle)
-            if ref:
-                try:
-                    EditRepository(self.dbstate,
-                               self.uistate, [], ref)
-                except WindowActiveError:
-                    pass
-        elif objclass == 'Note':
-            ref = self.dbstate.db.get_note_from_handle(handle)
-            if ref:
-                try:
-                    EditNote(self.dbstate,
-                             self.uistate, [], ref)
-                except WindowActiveError:
-                    pass
-        elif objclass == 'Media':
-            ref = self.dbstate.db.get_media_from_handle(handle)
-            if ref:
-                try:
-                    EditMedia(self.dbstate,
-                              self.uistate, [], ref)
-                except WindowActiveError:
-                    pass
-        elif objclass == 'Citation':
-            ref = self.dbstate.db.get_citation_from_handle(handle)
-            if ref:
+        if obj2class(objclass):  # make sure it is an editable object
+            if self.dbstate.db.method('has_%s_handle', objclass)(handle):
+                g_object = self.dbstate.db.method(
+                    'get_%s_from_handle', objclass)(handle)
                 try:
-                    EditCitation(self.dbstate, self.uistate, [], ref)
+                    locals()['Edit' + objclass](
```

Comment:
There is already an [EditObject](https://github.com/gramps-project/gramps/blob/master/gramps/gui/editors/__init__.py#L77) utility.  Could it be used here?

Reply:
@Nick-Hall I looked into using EditObject, but am not comfortable with the fact that it logs warnings when attempting to edit non-primary objects, or objects with invalid handles.  The original code was silent in these cases.  By the time I add in code to deal with these before calling EditObject, I have almost everything but the two line call you are commenting on.  I'd prefer to leave it as is.

BTW, an enhancement would be to allow editing of non-primary types, either here in the clipboard or elsewhere (like EditObject).  Then such a change would be more interesting.  But that is not the subject of this PR.

Reply:
OK.  I just thought that you may have been unaware of EditObject.